src/utils/safeHref.ts

```diff
@@ -0,0 +1,90 @@
+export interface URLInfo {
+    url: string | undefined
+    sameOrigin: boolean
+}
+
+let useSafeHref = false
+
+export function setUseSafeHref(value: boolean) {
+    useSafeHref = value
+}
+
+export function getUseSafeHref(): boolean {
+    return useSafeHref
+}
+
+export function getURLInfo(url: string): URLInfo {
+    /**
+     * An array of the safely allowed url protocols
+     */
+    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:', 'data:']
+
+    /**
+     * Create an anchor element to parse the protocol and origin of the url provided
+     */
+    const anchorElement = document.createElement('a')
```

Comment:
`document` doesn't exist on the server-side either. Might be a good idea to use something like https://github.com/Mikhus/domurl so it can work server-side.